### DISCUSSION TITLE
Avoid deploying latest version of a service as a default

### DISCUSSION THREAD

**Discussion by Edo78Carol** [mia-platform/community#127] [Ideas]
Recently we created a couple of services without creating a public variable to specify which version deploy in which environment.

The platform took the latest version as a default and while this may be useful in the dev environment having such a default in the preprod and prod envs show to be a risk because those two services were actually deployed for an oversight.

Luckily those services was just exposting test endpoints so no harm was done but if one of those services would be a deamon running periodically and writing in the DB is easy to imagine a dangerous scenario.

I think that having a way to specify which env should default to the latest version is the best solution. This will consent to automatically have the latest version as a default ONLY in the specified envs without risking to deploy a potentialy dangerous service in another env.

An easier and I hope faster alternative is to avoid a default version when the public variable isn't even defined. This way no rogue service can slip into an environment.

Is there some way, that I may be missing, to achieve this or to avoid human errors in this specific flow?

**Comment by fredmaggiowski** [comment]
Hi @Edo78Carol, thanks for the feedback.

Picking `latest` when no version is specified is actually the default behavior from docker (and therefore kubernetes) so there's little the Console can do right now apart from showing you that you're using the `latest` tags (when saving the configuration and when deploying it). 

If you want something embedded in Console I'd suggest you using Public Variables from the beginning to map different values based on environment and prevent such issues; otherwise you could edit your deployment pipeline to use tools like [conftest](https://github.com/open-policy-agent/conftest) and write your own policy for versions based on the environment you are deploying to so that the deploy on an environment different from test would fail.

Anyway thanks again for the feedback, we'll look into it and think what else we could to help users prevent these kind of problems @albertotessarotto @davidebianchi FYI

**Reply by davidebianchi** [reply]
Yes, I agree interesting feedback. We could think about checks to run on the configuration to deploy embedded into Console for the future.
At the moment, the suggestion of @fredmaggiowski it's a good solution for the usecase, we could add it in documentation to help users which have similar usecases.

**Comment by danibix95** [comment]
Hi @Edo78Carol, 

maybe, to avoid bringing unwanted changes in production, you can exploit multiple branches to work on. For example you can set up a branch for your non-production environments and one for releasing in production (e.g. _main_). When your work is ready to be release in production you can transfer your changes (either via merge request or replaying the changes on the Console) to the _main_ branch, tagging newer or updated services and creating a specific tag to be deployed on the cluster.

The [merge-configuration](https://docs.mia-platform.eu/docs/development_suite/api-console/api-design/merge_collaboration) feature, while still in beta, should also help in managing different branches.

P.S. in the commit dialog is already reported which services do not have a stable tag (e.g. `latest` or pre-release tag, such as `rc`)

**Comment by akelity** [comment]
This feature has received positive feedback and will be carefully evaluated for inclusion in the roadmap as soon as it becomes feasible.
